app/Http/Requests/User/Self/AbstractSelfEditRequest.php

```diff
@@ -0,0 +1,16 @@
+<?php
+
+namespace App\Http\Requests\User\Self;
+
+use App\Http\Requests\BaseApiRequest;
+use App\Models\User;
+use App\Policies\UserPolicy;
+use Illuminate\Support\Facades\Gate;
+
+abstract class AbstractSelfEditRequest extends BaseApiRequest
+{
+	public function authorize(): bool
+	{
+		return Gate::check(UserPolicy::CAN_EDIT_OWN_SETTINGS, User::class);
+	}
+}
```

Comment:
1. I am not sure if I really would go that far and create a class `AbstractSelfEditRequest` which only implements a single method and which is then used only at two places where you could just directly call `Gate::check(UserPolicy::CAN_EDIT_OWN_SETTINGS)` as well.
    This looks a little bit extreme and I fear that this will get soon out of hand up to an level where people cannot follow the code anymore. Moreover, it does not even save lines of code by pure numbers. In order to save 2x4 lines (two times the `authorize` method), you create a new file with 13 lines.
 2. Moreover, I still don't get why we need an extra right `UserPolicy::CAN_EDIT_OWN_SETTINGS`. IMHO, `UserPolicy::CAN_EDIT` still suffices.

Reply:
Refactored.